### DISCUSSION TITLE
Visual Studio Code Cleanup feature

### DISCUSSION THREAD

**Discussion by bstordrup** [dotnet/wpf#9610] [General]
Not sure if this is the right place to ask, but are contributors to this repository actively using the `Code Cleanup on Save` feature in Visual Studio 2022? And if so, what profile is used?

I'm asking because I'm experimenting with extending the `MessageBoxButton` and `MessageBoxResult` enumerations, and because of the `Code Cleanup on Save`, I'm getting a bunch of unrelated changes caused by the code cleanup.

I could of course turn if off, but I'm happy for it in other repositories, and I'm unsure if the setting is somehow stored in the repository.

**Comment by miloush** [comment]
I am not using that feature and I am unhappy seeing PRs cluttered with formatting changes together with functional changes. WPF is a big codebase written over many years, with different styles in different parts and I would generally follow the style in whatever file I am editing.

**Reply by lindexi** [reply]
Long long ago, I use [dotnet format](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-format) tool to format the wpf repo...

**Comment by h3xds1nz** [comment]
I'm not using it either but I still make miloush unhappy. By the way, do note that extending those enumerations etc. would require to go through the API Review process for .NET in order for those to be approved as a shape.

**Reply by miloush** [reply]
Fortunately it is not a requirement for me to be happy. :-) It makes it easier to review though.

**Reply by bstordrup** [reply]
@h3xds1nz Thank you. API Review process - is that documented in this repository, or is it a different repository I need to look into?

**Reply by miloush** [reply]
It is the same (this) repo. If you create a new issue, there should be an API suggestion template. You can still create your PR but getting an API approval would be a prerequisite for merging it.

**Reply by bstordrup** [reply]
OK. I'll create a new issue and link #5795 in it.

**Comment by bstordrup** [comment]
Just for the record - the experimenting on new `MessageBoxButtons` values are in connection with issue #5795.
